bundles/org.codechecker.eclipse.plugin/src/org/codechecker/eclipse/plugin/config/global/CcGlobalConfiguration.java

```diff
@@ -0,0 +1,106 @@
+package org.codechecker.eclipse.plugin.config.global;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.codechecker.eclipse.plugin.CodeCheckerNature;
+import org.codechecker.eclipse.plugin.Logger;
+import org.codechecker.eclipse.plugin.config.CcConfigurationBase;
+import org.codechecker.eclipse.plugin.config.Config.ConfigLogger;
+import org.codechecker.eclipse.plugin.config.Config.ConfigTypes;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.preferences.ConfigurationScope;
+import org.osgi.service.prefs.BackingStoreException;
+
+import com.google.common.collect.Sets;
+
+/**
+ * This Class represents the global configuration of the plugin.
+ */
+public class CcGlobalConfiguration extends CcConfigurationBase {
+    private static final String STR_EMPTY = "";
+
+    private static CcGlobalConfiguration instance;
+
+    /**
+     * Hidden due singleton pattern.
+     */
+    private CcGlobalConfiguration() {}
+
+    /**
+     * @return This instance.
+     */
+    public static CcGlobalConfiguration getInstance() {
+        if (instance == null)
+            instance = new CcGlobalConfiguration();
+        return instance;
+
+    }
+
+    /**
+     * Initializes the plug-ins internal configuration, Call this method somewhere early.
```

Comment:
What does it mean early? Couldn't we call it automatically, after initialization somewhere?

Reply:
Thanks, I'll call this from the instance retrivial method. From the constructor branch, and remove from startup.